readthedocs/doc_builder/base.py

```diff
@@ -94,37 +100,46 @@ def docs_dir(self, docs_dir=None, **__):
             docs_dir = checkout_path
         return docs_dir
 
-    def create_index(self, extension='md', **__):
-        """Create an index file if it needs it."""
+    def create_index(self, extension='md', force_index=False, **__):
+        """
+        Create an index file if it needs it.
+
+        If force_index is True and there isn't an index file,
+        one is created whether or not there is a README file.
+        """
         docs_dir = self.docs_dir()
 
         index_filename = os.path.join(
-            docs_dir, 'index.{ext}'.format(ext=extension))
-        if not os.path.exists(index_filename):
-            readme_filename = os.path.join(
-                docs_dir, 'README.{ext}'.format(ext=extension))
-            if os.path.exists(readme_filename):
-                return 'README'
-
-            index_file = open(index_filename, 'w+')
-            index_text = """
-
-Welcome to Read the Docs
-------------------------
-
-This is an autogenerated index file.
-
-Please create an ``index.{ext}`` or ``README.{ext}`` file with your own content
-under the root (or ``/docs``) directory in your repository.
-
-If you want to use another markup, choose a different builder in your settings.
-Check out our `Getting Started Guide
-<https://docs.readthedocs.io/en/latest/getting_started.html>`_ to become more
-familiar with Read the Docs.
-                """
-
-            index_file.write(index_text.format(dir=docs_dir, ext=extension))
-            index_file.close()
+            docs_dir, 'index.{ext}'.format(ext=extension)
+        )
+        readme_filename = os.path.join(
+            docs_dir, 'README.{ext}'.format(ext=extension)
+        )
+        if os.path.exists(index_filename):
+            return 'index'
+        if not force_index and os.path.exists(readme_filename):
+            return 'README'
+
+        index_text = textwrap.dedent(
+            """
+            Welcome to Read the Docs
+            ------------------------
+
+            This is an autogenerated index file.
+
+            Please create an ``index.{ext}`` or ``README.{ext}`` (Sphinx only) file with
+            your own content under the root (or ``/docs``) directory in your repository.
+
+            If you want to use another markup, choose a different builder in your settings.
+            Check out our `Getting Started Guide
+            <https://docs.readthedocs.io/en/latest/getting_started.html>`_ to become more
+            familiar with Read the Docs.
+            """
+        )
```

Comment:
I also realize that this will look weird on mkdocs (because of the rst specific format used here)